docs/ui/components/Text/index.tsx

```diff
@@ -0,0 +1,169 @@
+import { css, SerializedStyles } from '@emotion/react';
+import { theme, typography } from '@expo/styleguide';
+import React, { HTMLAttributes } from 'react';
+
+import { Link, LinkProps } from '~/ui/components/Link';
+import { durations } from '~/ui/foundations/durations';
+
+export enum TextElement {
+  CODE = 'code',
+  H1 = 'h1',
+  H2 = 'h2',
+  H3 = 'h3',
+  H4 = 'h4',
+  H5 = 'h5',
+  H6 = 'h6',
+  LI = 'li',
+  P = 'p',
+  SPAN = 'span',
+  UL = 'ul',
+  OL = 'ol',
+  PRE = 'pre',
+}
+
+type TextComponentProps = HTMLAttributes<
+  HTMLHeadingElement | HTMLParagraphElement | HTMLLIElement | HTMLUListElement | HTMLPreElement
+> & {
+  testID?: string;
+  size?: 'small';
+  weight?: 'regular' | 'medium' | 'semiBold' | 'black';
+  theme?: 'secondary' | 'success' | 'warning' | 'error';
```

Comment:
@jonsamp how do you feel about linking this with the Expo styleguide more? E.g.

```suggestion
  weight?: keyof typeof typography.utility.weight;
  theme?: keyof typeof theme.text;
```

This should auto-update if we publish or change the styleguide, making it a bit more type-safe. With this, we can also simplify `getThemeStyle` and `getWeightStyle`, and just pipe through what ever we have in the styleguide.

Reply:
Done!